Sfork issu 182 (#216)
numpy/numpy.org: content/en/case-studies/gw-discov.md

gravitational waves. Each of them has multi-kilometer-scale gravitational wave
detectors that use laser interferometry.  The LIGO Scientific Collaboration
(LSC), is a group of more than 1000 scientists from universities around the
United States and in 14 other countries supported by more than 90 universities
and research institutes; approximately 250 students actively contributing to the
collaboration. LIGO discovery GW150914 is the first observation that is
significant in being the first observation both confirming the existence of
gravitational waves as well as proves the ability to directly measure them. 

Gravitational waves allow scientists to observe and study the universe in a
brand new way, ushering in the era of gravitational wave astronomy.

### Key Objectives

* Though its [mission](https://www.ligo.caltech.edu/page/what-is-ligo) is to
  detect gravitational waves from some of the most violent and energetic
processes in the Universe, the data LIGO collects may have far-reaching effects
on many areas of physics including gravitation, relativity, astrophysics,
cosmology, particle physics, and nuclear physics.

> * Crunch observed data via numerical relativity computations that involves
>   complex maths in order to discern signal from noise, filter out relevant

Reviewer: ```suggestion
* Analyze the raw interferometry data, which involves computation and
   analysis techniques from a wide variety of fields, including physics,
   signal processing, and statistics.
```
The original wording is a bit misleading in that it conflates the numerical relativity computations with signal processing.

signal and statistically estimate significance of observed data 

> * Data visualization so that the binary / numerical results can be
>   comprehended.

Reviewer: ```suggestion
* Data visualization so that the numerical results can be
  comprehended.
```
I'm not sure about this bullet. Visualization is nice, but I wouldn't count is as a key objective, nor is the main purpose necessarily comprehension. Perhaps rephrasing to emphasize the generation of high-level results suitable for comparing with model predictions and visualization.

* **Computation**

    Gravitational Waves are hard to detect as they produce a very small effect
and have tiny interaction with matter.  Processing and analyzing all of LIGO's
data requires a vast computing infrastructure.After taking care of noise, which
is billions of times of the signal, there is still very complex relativity
equations and huge amounts of data which present a computational challenge:
[O(10^7) CPU hrs needed for binary merger
analyses](https://youtu.be/7mcHknWWzNI) spread on 6 dedicated LIGO clusters 

* **Data Deluge**

    As observational devices become more sensitive and reliable, the challenges
posed by data deluge and finding a needle in a haystack rise multi-fold.  LIGO
generates terabytes of data every day! Making sense of this data requires an
enormous effort for each and every detection. For example, the signals being
collected by LIGO must be matched by supercomputers against hundreds of
thousands of templates of possible gravitational-wave signatures. 

* **Visualization**

>     Once the obstacles related to understanding Einstein’s equations well enough
> to solve them using supercomputers are taken care of, the next big challenge was
> making data comprehensible to the human brain. Simulation modeling as well as

Reviewer: ```suggestion
Simulation modeling as well as
```
This statement misses the mark a bit and is not really necessary. I agree that highlighting the role of visualization is nice though.

> signal detection requires effective visualization techniques.  Visualization
> also plays a role in lending more credibility to numerical relativity in the
> eyes of pure science aficionados, who did not give enough importance to
> numerical relativity until imaging and simulations made it easier to comprehend
> results for a larger audience.  Speed of complex computations and rendering,

Reviewer: ```suggestion
signal detection requires effective visualization techniques.
  Speed of complex computations and rendering,
```
Similar to the leading sentence - this is not necessarily true (scientists don't rely on visualization for confirmation of their findings - more important are the statistics that gauge the agreement between model predictions and observations.

re-rendering images and simulations using latest experimental inputs and
insights can be a time consuming activity that challenges researchers in this
domain.

{{< figure src="/images/content_images/cs/gw_measurement.png" 
    class="fig-center" alt="gravitational waves measurement" 
    caption="**LIGO measurement of gravitational waves at Hanford and Livingston comparing signals to those expected due to a black hole merger event**" 

>     attr="(**Graph Credits:** LIGO Scientific Collboration & Virgo Collaboration" 

Reviewer: ```suggestion
    attr="(**Graph Credits:** LIGO Scientific Collboration & Virgo Collaboration)" 
```

    attrlink="https://physics.aps.org/featured-article-pdf/10.1103/PhysRevLett.116.061102" >}}

## NumPy’s Role in the detection of Gravitational Waves
	
Gravitational waves emitted from the merger cannot be computed using any

> technique except brute force numerical relativity using supercomputers.  The
> amount of data LIGO collects is as incomprehensibly large and requires careful
> processing like any other data science problem.  Although it is technologically
> more challenging to achieve the requisite sensitivity for capturing
> gravitaional waves and is way harder than dealing with large scale data, yet
> the latter is complex enough to require numerical processing skills of several
> Python tools including
> [NumPy](https://towardsdatascience.com/2020-how-a-i-could-help-astronomers-sorting-big-data-811571705707).

Reviewer: ```suggestion
```
We should be careful mentioning NumPy in the context of the template matching and relativity calculations. I'm not sure that NumPy was used in those regimes.

My suggestion would be to delete this paragraph, the following paragraph on NumPy doesn't need any sort of segue, and I think the statements in this one are misleading and/or inaccurate.

I see that you have a link to a towardsdatascience article - maybe this section was included for the "honeypot" aspect. I must say though that the data/tutorials from PyCBC are much more compelling than the one in this article (which uses pre-processed data that doesn't really use NumPy except for loading the data).

NumPy, the standard numerical analysis package for Python,  was utilized by the
software used for various tasks performed during the GW detection project at

> LIGO. NumPy helped in solving complex maths and data manipulation at high speed.
> Here are some examples:

Reviewer: ```suggestion
LIGO. Here are some examples:
```

* **[Signal Processing](https://www.uv.es/virgogroup/Denoising_ROF.html):**  Glitch
  detection,  [Noise identification and Data
Characterization](https://ep2016.europython.eu/media/conference/slides/pyhton-in-gravitational-waves-research-communities.pdf)
(NumPy, scikit-learn, scipy, matplotlib, pandas, pyCharm )
* **GW data analysis:**
  [GwPy](https://gwpy.github.io/docs/stable/overview.html) and

> [PyCBC](https://pycbc.org) are two of the [key LIGO experimentation 
> softwares](https://github.com/lscsoft) 

Reviewer: ```suggestion
[PyCBC](https://pycbc.org) are two [LIGO data analysis
packages](https://github.com/lscsoft) 
```

> that use NumPy and AstroPy under the hood, for providing

Reviewer: ```suggestion
that use NumPy and AstroPy under the hood, providing
```

> object based interfaces to utilities, tools and methods for studying data from

Reviewer: ```suggestion
object-based interfaces to utilities, tools and methods for studying data from
```

> gravitational-wave detectors. Both GwPy and pycbc libraries rely on NumPy for

Reviewer: ```suggestion
gravitational-wave detectors. Both GwPy and PyCBC libraries rely on NumPy for
```

> analysis of the "chirp" signal. For example, in pycbc, the time series data are
> stored in ndarrays and scipy.signal, which operates on NumPy arrays.

Reviewer: ```suggestion
analysis of the "chirp" signal. For example, in PyCBC, the time series data are
stored in `ndarrays` and `scipy.signal`, which operates on NumPy arrays, is
used for digital signal processing.
```

* Data retrieval: Deciding which data can be analyzed, figuring out whether it
  contains a signal - needle in a haystack
* Statistical analysis: estimate the statistical significance of observational

>   data, estimating the signal parameters (e.g. masses of stars, spin velocity,

Reviewer: These bullets seem incomplete - are they notes-to-self or intended to be included? If the latter, I think they need to be developed more, though I think the above bullet about GwPy and PyCBC adequately covers components like this.

and distance) by comparison with a model.
* Visualization of data
  - Time series

>   - Spectrograms

Reviewer: Same comment as above

## Summary

> GW detection has enabled researchers to discover entirely unexpected phenomena
> while providing new insight into many of the most profound astrophysical

Reviewer: ```suggestion
GW detection has enabled researchers to observe the universe in a brand new way,
providing new insight into many of the most profound astrophysical
```

> phenomena known. Number crunching and data visualization is a crucial step that
> helps scientists gain insights into data gathered from the scientific
> observations and understand the results. The computations are complex and cannot

Reviewer: ```suggestion
phenomena known. Data analysis and visualization play a crucial role in
helping scientists interpret data gathered from the scientific
observations and communicate the results.
```

> be comprehended by humans unless it is visualized using computer simulations
> that are fed with the real observed data and analysis.  NumPy along with other

Reviewer: ```suggestion
NumPy along with other
```
Again, this point is not correct.
